lib/rules/no-template-curly-in-string.js

```diff
@@ -22,23 +22,43 @@ module.exports = {
         schema: [],
 
         messages: {
-            unexpectedTemplateExpression: "Unexpected template string expression."
-        }
+            unexpectedTemplateExpression: "Unexpected template string expression.",
+            convertToTemplate: "Convert to ES6 `template`."
+        },
+        hasSuggestions: true
     },
 
     create(context) {
         const regex = /\$\{[^}]+\}/u;
+        const sourceCode = context.getSourceCode();
 
         return {
             Literal(node) {
                 if (typeof node.value === "string" && regex.test(node.value)) {
                     context.report({
                         node,
-                        messageId: "unexpectedTemplateExpression"
+                        messageId: "unexpectedTemplateExpression",
+                        suggest: [
+                            {
+                                messageId: "convertToTemplate",
+                                fix: fixer => {
+                                    const text = sourceCode.getText(node);
+                                    const oldQuote = text[0];
+                                    const textNoQuotes = text.slice(1, -1);
+                                    const escapeBackticks = textNoQuotes.replaceAll("`", "\\`");
+                                    const unescapeQuotes = escapeBackticks.replaceAll(`\\${oldQuote}`, oldQuote);
+                                    // eslint-disable-next-line require-unicode-regexp -- Needs fixing, adding the unicode flag results in "Incomplete quantifier" parsing error
+                                    const backTicksInsideCurly = /(?<=\${)\\`([^`]*)\\`(?=})/gi;
```

Comment:
```suggestion
                                    const backTicksInsideCurly = /(?<=\$\{)\\`([^`]*)\\`(?=\})/gi;
```
seems you need to escape `{` and `}`.

Reply:
Got it, thank you very much!